src/test/scala/example/ServiceTest.scala

```diff
@@ -0,0 +1,301 @@
+package example
+
+import org.scalatest.{FlatSpec, Matchers}
+import cats.implicits.catsStdInstancesForFuture
+import cats.Id
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+
+class ServiceTest extends FlatSpec with Matchers {
+
+  "The user repository id method calls" should "return valid result" in {
+    val userRepositoryId = new UserRepositoryId()
+
+    val johnUsername = "John Smith"
+    val johnId = 1
+
+    val registeredUser = userRepositoryId.registerUser(johnUsername)
+    registeredUser.id shouldEqual johnId
+    registeredUser.username shouldEqual johnUsername
+
+    val retrievedByUsername = userRepositoryId.getByUsername(johnUsername).get
+    retrievedByUsername.id shouldEqual johnId
+    retrievedByUsername.username shouldEqual johnUsername
+
+    val retrievedById = userRepositoryId.getById(johnId).get
+    retrievedById.id shouldEqual johnId
+    retrievedById.username shouldEqual johnUsername
+
+    val breadUsername = "Bread Pitt"
+    val breadId = 2
+    userRepositoryId.registerUser(breadUsername).id shouldEqual breadId
+
+    userRepositoryId.getById(3) shouldBe Option.empty
+    userRepositoryId.getByUsername("Random user name") shouldBe Option.empty
+
+    val john2Username = "John Smith"
+    val john2Id = 3
+
+    // user will not be registered because there is one with such username
+    userRepositoryId.registerUser(john2Username).id shouldEqual johnId
+    userRepositoryId.getById(john2Id) shouldEqual Option.empty
+  }
+
+  "The user repository future method calls" should "return valid result" in {
+    val userRepositoryFuture = new UserRepositoryFuture()
+
+    val johnUsername = "John Smith"
+    val johnId = 1
+
+    val registeredUser = Await.result(userRepositoryFuture.registerUser(johnUsername), 2.seconds)
```

Comment:
consider please also http://www.scalatest.org/user_guide/async_testing instead of blocking